src/ilasm/assembler.cpp

```diff
@@ -126,8 +126,16 @@ mdToken Assembler::ResolveClassRef(mdToken tkResScope, __in __nullterminated con
             }
             return tkRet;
         }
-        else  // needs to be resolved
-            if(!m_fIsMscorlib) tkResScope = GetBaseAsmRef();
+        else
+        { // needs to be resolved
+            if (TypeFromToken(tkResScope) == mdtAssemblyRef && RidFromToken(tkResScope) > 0)
+            {
+                // considering the aliased case '[' dottedName ']' slashedName
+                AsmManAssembly* assembly = m_pManifest->m_AsmRefLst.PEEK(RidFromToken(tkResScope) - 1);
```

Comment:
1. Sure, the first condition  "TypeFromToken(tkResScope) == mdtAssemblyRef" guarantees this fact.
2. I do not know. I am not familiar with ILAsm either. If there is someone who knows please help us.